### PR TITLE
DDF-4047 refactor security core common to better support SOAP clients

### PR DESCRIPTION
**Pull request by stustison**
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@coyotesqrl
@pklinef
@stustison
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by stustison** [comment]
build now

**Comment by clockard** [comment]
build now

File: platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java [hunk 1]
```diff
@@ -155,17 +162,19 @@ private SecurityToken getSecurityToken(WebServiceProperties wsp, Serializable se
...
+              try {
+                return clientFactory.create(wsp.serviceClass);
+              } catch (ServiceConstructionException e) {
+                LOGGER.debug(
+                    "Unable to use WSDL to build client. Attempting to use service class.", e);
+                unavailableWsdls.add(wsp.endpointWsdlURL);
+                clientFactory.getClientFactoryBean().getServiceFactory().setPopulateFromClass(true);
+                return clientFactory.create(wsp.serviceClass);
+              }
+            });
```
Comment (brjeter) [inline platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java]: ❓ In the future, do we plan to check a custom permission before doing this?

Comment (stustison) [reply platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java]: Yes, I'll probably add something just like I did for the REST client.

File: platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java [hunk 2]
```diff
@@ -155,17 +162,19 @@ private SecurityToken getSecurityToken(WebServiceProperties wsp, Serializable se
...
-    }
-
-    return proxyServiceType;
+    return AccessController.doPrivileged(
+        (PrivilegedAction<ProxyServiceType>)
+            () -> {
+              try {
+                return clientFactory.create(wsp.serviceClass);
+              } catch (ServiceConstructionException e) {
+                LOGGER.debug(
```
[the remark this answers is not in the document]
Comment (pklinef) [reply platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java]: Adding the context that `ServiceConstructionException` is `RuntimeException`.
https://cxf.apache.org/javadoc/latest/org/apache/cxf/service/factory/ServiceConstructionException.html

[the remark this answers is not in the document]
Comment (coyotesqrl) [reply platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java]: Well that does make a significant difference. Should have checked that. Thanks, @pklinef

**Pull request merged**